components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -202,21 +202,19 @@ public void connect() throws ConnectionUnavailableException {
 
             //initializing SMTP server to create session object.
             if (smtpUsername != null && smtpPassword != null) {
-                session = Session.getInstance(props, new Authenticator() {
-                    public PasswordAuthentication getPasswordAuthentication() {
-                        return new PasswordAuthentication(smtpUsername, smtpPassword);
-                    }
-                });
+                if (!smtpUsername.isEmpty() && !smtpPassword.isEmpty()) {
+                    session = Session.getInstance(props, new Authenticator() {
+                        public PasswordAuthentication getPasswordAuthentication() {
+                            return new PasswordAuthentication(smtpUsername, smtpPassword);
+                        }
+                    });
+                } else {
+                    session = Session.getInstance(props);
+                }
             } else {
-                log.error("Error in smtp username & password verification");
-                String msg = "failed to connect to the mail server due to failed " +
-                        "user password authorization";
-                throw new ConnectionUnavailableException("The adapter " +
-                        eventAdapterConfiguration.getName() + " " + msg);
+                session = Session.getInstance(props);
```

Comment:
@thiliA 
Do we need to handle the else case for username, password being null separately and the case for username, password being empty separately? Can you re-arrange the code so that the same code segment 'session = Session.getInstance(props);' does not get repeated?
Also, if we are connecting without using username, password, I think it would be better to add a log saying that we are connecting without username, password. WDYT?